transaction-status-client-types/src/lib.rs

```diff
@@ -965,4 +973,28 @@ mod test {
                 .expect("Failed to deserialize `UiTransactionError");
         assert_eq!(actual_transaction_error, expected_transaction_error);
     }
+
+    #[test]
+    fn test_deserialize_instruction_error_string_format() {
+        // Test that we can deserialize new InstructionErrors when serialized as
+        // a string.
+        let new_error = TransactionError::InstructionError(0, InstructionError::BorshIoError);
+        let error_json = to_value(&new_error).unwrap();
+        let result = from_value::<UiTransactionError>(error_json);
+        assert!(matches!(
+            result.unwrap().0,
+            TransactionError::InstructionError(0, InstructionError::BorshIoError)
+        ));
+
+        // This checks compatibility across SDK versions where BorshIoError is
+        // a unit variant in v3 and newtype variant in v2.
+        let old_error =
+            to_value(serde_json::json!({"InstructionError": [0, {"BorshIoError": "Unknown"}]}))
```

Comment:
won't this lend to breakage should we wind up with a typo here?

Reply:
I take your point that we don't get a compile-error from a misspelling here, which we would for `InstructionError::BorshIoError`. On the flip-side, if the test starts failing because we made a typo, that means the test is doing its job, right?

We'd only get a false positive if we change this string, the one on line 316, but *not* the one on line 317. While that's possible, I'd argue that it's very unlikely. What do you think?